### PR TITLE
Removed CPEX from Communicator.

### PR DESCRIPTION
**Pull request by heplesser**
This PR removes the CPEX algorithm from communicator, partially solving old ticket no 364.

### CODE REVIEW

**Comment by abigailm** [comment]
It looks fine to me, +1 to merge

File: nestkernel/communicator.cpp [hunk 1]
```diff
@@ -261,10 +224,8 @@ nest::Communicator::communicate( std::vector< uint_t >& send_buffer,
     }
     recv_buffer.swap( send_buffer );
   }
-  else if ( ( num_processes_ > 1 ) && use_Allgather_ ) // communicate using Allgather
+  else 
     communicate_Allgather( send_buffer, recv_buffer, displacements );
```
Comment (wschenck) [inline nestkernel/communicator.cpp]: I would suggest curly brackets around the else statements.

File: nestkernel/communicator.cpp [hunk 2]
```diff
@@ -527,10 +374,8 @@ nest::Communicator::communicate( std::vector< OffGridSpike >& send_buffer,
...
```
Comment (wschenck) [inline nestkernel/communicator.cpp]: I would suggest curly brackets around the else statements.

**Comment by wschenck** [comment]
Overall it looks fine to me, but see my small comments.

And I found further references to the communicate_allgather status variable in the following files:
* examples/nest/Potjans_2014/microcircuit.sli (line 112)
* examples/nest/Potjans_2014/sim_params.sli (line 40)

**Comment by heplesser** [comment]
@wschenck Thanks for your comments, I am working on it!

**Comment by wschenck** [comment]
@apeyser : The pull request looks fine, I would suggest to merge it in.

**Pull request merged**